### ISSUE TITLE
rpc ERROR - unable to read database

### ISSUE THREAD

**Issue by RochmanReese** [iquidus/explorer#496]
Hi,

I have things mostly set up now and have been trying for hours to find out why this error is happening.

I have checked and rechecked all my rpc settings and database settings and they all appear to match. 

The web GUI comes up and look great, however has no block data showing.

I have tried reinstalled on a a fresh ubuntu 1804 server and still am getting the same problem.
no block data showing in the explorer.

Mongo 4.2.22
node v12.14.0

Any help gratefully appreciated.

when i click on the web page i can see the GET call happening instantly:-

GET /ext/getlasttxsajax/0?draw=358&columns%5B0%5D%5Bdata%5D=0&columns%5B0%5D%5Bname%5D=&columns%5B0%5D%5Bsearchable%5D=true&columns%5B0%5D%5Borderable%5D=false&columns%5B0%5D%5Bsearch%5D%5Bvalue%5D=&columns%5B0%5D%5Bsearch%5D%5Bregex%5D=false&columns%5B1%5D%5Bdata%5D=1&columns%5B1%5D%5Bname%5D=&columns%5B1%5D%5Bsearchable%5D=true&columns%5B1%5D%5Borderable%5D=false&columns%5B1%5D%5Bsearch%5D%5Bvalue%5D=&columns%5B1%5D%5Bsearch%5D%5Bregex%5D=false&columns%5B2%5D%5Bdata%5D=2&columns%5B2%5D%5Bname%5D=&columns%5B2%5D%5Bsearchable%5D=true&columns%5B2%5D%5Borderable%5D=false&columns%5B2%5D%5Bsearch%5D%5Bvalue%5D=&columns%5B2%5D%5Bsearch%5D%5Bregex%5D=false&columns%5B3%5D%5Bdata%5D=3&columns%5B3%5D%5Bname%5D=&columns%5B3%5D%5Bsearchable%5D=true&columns%5B3%5D%5Borderable%5D=false&columns%5B3%5D%5Bsearch%5D%5Bvalue%5D=&columns%5B3%5D%5Bsearch%5D%5Bregex%5D=false&columns%5B4%5D%5Bdata%5D=4&columns%5B4%5D%5Bname%5D=&columns%5B4%5D%5Bsearchable%5D=true&columns%5B4%5D%5Borderable%5D=false&columns%5B4%5D%5Bsearch%5D%5Bvalue%5D=&columns%5B4%5D%5Bsearch%5D%5Bregex%5D=false&start=0&length=10&search%5Bvalue%5D=&search%5Bregex%5D=false&_=1661409089833 200 7.109 ms - 61
GET /ext/summary 304 24.711 ms - -
GET / 304 604.452 ms - -
GET /themes/Lumen/bootstrap.min.css 304 1.273 ms - -
GET /vendor/flags-css/css/flag-icon.min.css 304 0.946 ms - -
GET /stylesheets/style.css 304 1.163 ms - -
GET /vendor/jqplot/jquery.jqplot.css 304 0.464 ms - -
GET /vendor/jqplot/jquery.jqplot.min.js 304 0.317 ms - -
GET /vendor/jqplot/plugins/jqplot.dateAxisRenderer.min.js 304 0.378 ms - -
GET /vendor/jqplot/plugins/jqplot.ohlcRenderer.min.js 304 0.450 ms - -
GET /vendor/jqplot/plugins/jqplot.highlighter.min.js 304 0.771 ms - -
GET /vendor/jqplot/plugins/jqplot.pieRenderer.min.js 304 0.351 ms - -
GET /vendor/jqplot/plugins/jqplot.barRenderer.min.js 304 0.755 ms - -
GET /vendor/jqplot/plugins/jqplot.categoryAxisRenderer.min.js 304 0.490 ms - -
GET /javascripts/chart.js 304 0.294 ms - -
GET /images/logo.png 304 0.664 ms - -
GET /ext/getlasttxsajax/0?draw=1&columns%5B0%5D%5Bdata%5D=0&columns%5B0%5D%5Bname%5D=&columns%5B0%5D%5Bsearchable%5D=true&columns%5B0%5D%5Borderable%5D=false&columns%5B0%5D%5Bsearch%5D%5Bvalue%5D=&columns%5B0%5D%5Bsearch%5D%5Bregex%5D=false&columns%5B1%5D%5Bdata%5D=1&columns%5B1%5D%5Bname%5D=&columns%5B1%5D%5Bsearchable%5D=true&columns%5B1%5D%5Borderable%5D=false&columns%5B1%5D%5Bsearch%5D%5Bvalue%5D=&columns%5B1%5D%5Bsearch%5D%5Bregex%5D=false&columns%5B2%5D%5Bdata%5D=2&columns%5B2%5D%5Bname%5D=&columns%5B2%5D%5Bsearchable%5D=true&columns%5B2%5D%5Borderable%5D=false&columns%5B2%5D%5Bsearch%5D%5Bvalue%5D=&columns%5B2%5D%5Bsearch%5D%5Bregex%5D=false&columns%5B3%5D%5Bdata%5D=3&columns%5B3%5D%5Bname%5D=&columns%5B3%5D%5Bsearchable%5D=true&columns%5B3%5D%5Borderable%5D=false&columns%5B3%5D%5Bsearch%5D%5Bvalue%5D=&columns%5B3%5D%5Bsearch%5D%5Bregex%5D=false&columns%5B4%5D%5Bdata%5D=4&columns%5B4%5D%5Bname%5D=&columns%5B4%5D%5Bsearchable%5D=true&columns%5B4%5D%5Borderable%5D=false&columns%5B4%5D%5Bsearch%5D%5Bvalue%5D=&columns%5B4%5D%5Bsearch%5D%5Bregex%5D=false&start=0&length=10&search%5Bvalue%5D=&search%5Bregex%5D=false&_=1661430550651 200 11.527 ms - 59
GET /ext/summary 304 29.219 ms - -

shows it is running the correct db user and connecting ok:-

2022-08-25T13:25:41.979+0100 I  ACCESS   [conn1] Successfully authenticated as principal fucuexplorer on explorer from client 127.0.0.1:54472
2022-08-25T13:25:42.034+0100 I  NETWORK  [listener] connection accepted from 127.0.0.1:54476 #2 (2 connections now open)
2022-08-25T13:25:42.036+0100 I  NETWORK  [conn2] received client metadata from 127.0.0.1:54476 conn2: { driver: { name: "nodejs-core", version: "3.3.3" }, os: { type: "Linux", name: "linux", architecture: "x64", version: "4.15.0-20-generic" }, platform: "Node.js v12.14.0, LE" }

**Comment by joeuhren** [comment]
What is the error or problem you are having exactly?  If it's just that you don't see any data, have you tried running the main sync cmd from within the explorer directory? `/path/to/node scripts/sync.js index update`

**Comment by RochmanReese** [comment]
Hi, thanks for your quick response.

I now realise that I have misnamed the title of the problem. It is unlikely to be a RPC error as everything seems to be connecting ok.

The problem is that I am not seeing any of the block date on the explorer web page.
if I enter into terminal
/root/explorer/scripts/sync.js index update
i then get this error message

./sync.js: line 1: syntax error near unexpected token `('
./sync.js: line 1: `var mongoose = require('mongoose')'

if I cd into the folder and run ./sync.js index update I get same error message

Thanks for your help

**Comment by joeuhren** [comment]
The sync cmd is pretty particular about how it is called, and it seems you are calling it incorrectly. Here are 2 tips:

1. You need to be inside the root of the explorer directory when you run the cmd
2. The script must be run with node.js, so you must specify the path to node.js before the cmd

Try this:

```
cd /root/explorer
/path/to/node scripts/sync.js index update
```

Be sure to replace `/path/to/node` with the actual full path to node.js. If you don't know the full path to where you installed node.js, try running `which node` and it should show you the full path that you need to use with the cmd above.

**Comment by RochmanReese** [comment]
Thank you so much for your clear instructions in how to fix the problem.
Everything is working well now. 
I changed the cron files to
*/1 * * * * cd /root/explorer && /root/.nvm/versions/node/v12.14.0/bin/node scripts/sync.js index update > /dev/null 2>&1
*/2 * * * * cd /root/explorer && /root/.nvm/versions/node/v12.14.0/bin/node scripts/market.js > /dev/null 2>&1
*/5 * * * * cd /root/explorer && /root/.nvm/versions/node/v12.14.0/bin/node scripts/peers.js > /dev/null 2>&1

seems to be working.

Thanks again for your help

**Comment by RochmanReese** [comment]
Hi,
I rebooted the system to see what needed to be done on reboot to keep explorer runnning.

Unfortunately it now no longer works. 
When I run npm start it just hangs with this message -
> explorer@1.7.4 start /root/explorer
> node --stack-size=10000 ./bin/cluster

Starting cluster with pid: 1626

the webpage keeps returning - 

This site can’t be reached192.210.233.100 refused to connect.

I have checked that mongod is running 
my coind is running
mongo database is set to explorer
mongo can see my database

thanks for your help.
